source/common/router/router.h

```diff
@@ -276,6 +310,11 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   uint32_t buffer_limit_{0};
   bool stream_destroyed_{};
 
+  // list of cookies to add to upstream headers
+  // 'mutable' is safe because this is only read in onUpstreamHeaders(), which
+  // is not const
```

Comment:
I believe this tuple is key,value,ttl, but using the tuple makes this hard to see. You could consider using a named type (like a struct) instead as an option, but if not I think a comment here naming the fields of the tuple would be a good idea. In C++14 you can create a sort of generic named/tagged tuple type, maybe some day we can use that.

Reply:
What was the reason for `list` vs. `vector`?